### PR TITLE
osbuilder/scripts: add support to yq version 4 and above

### PR DESCRIPTION
**Pull request by fgiudici**
yq changed syntax in an incompatible way starting from version 4 and
above. Add support to all versions.

Fixes https://github.com/kata-containers/kata-containers/issues/2297

I think this should fix the snap building process in the CI

### CODE REVIEW

**Comment by Jakob-Naucke** [comment]
I'm not against this PR, but I do wonder how yq 4 was used with this https://github.com/kata-containers/kata-containers/blob/b817340f94f050c41c743bac1e601ce38a3050b7/ci/install_yq.sh#L21

**Comment by fgiudici** [comment]
> I'm not against this PR, but I do wonder how yq 4 was used with this
> 
> https://github.com/kata-containers/kata-containers/blob/b817340f94f050c41c743bac1e601ce38a3050b7/ci/install_yq.sh#L21

I think because to read the musl and rust versions we would use yq installed on the host first if available. Here: 
https://github.com/kata-containers/kata-containers/blob/main/tools/osbuilder/scripts/lib.sh#L341

**Review by devimc — APPROVED** [review]
thanks @fgiudici

**Comment by Jakob-Naucke** [comment]
@devimc that didn't pick up for whatever reason -- let me try again
/test

**Comment by fgiudici** [comment]
> @devimc ~that didn't pick up for whatever reason -- let me try again~ didn't notice that force push
> /test
Yep, my fault... I corrected a couple of trivial mistakes :-P
Thanks for starting the tests

**Review by Jakob-Naucke — COMMENTED** [review]
@fguidici no worries, 2 updates is nothing :)
I wonder whether we should use this in more places where we use yq:
```
$ grep -rEI 'yq[\}"]? r'                          
src/runtime/golang.mk:    golang_version_min=$(shell $(GOPATH)/bin/yq r ../../versions.yaml languages.golang.version)
src/runtime/virtcontainers/pkg/cloud-hypervisor/Makefile:	clh_version=$(shell yq r $(VERSIONS_FILE) assets.hypervisor.cloud_hypervisor.version); \
tools/osbuilder/scripts/lib.sh:	RUST_VERSION="$(cat "${kata_versions_file}" | $yq r -X - "languages.rust.meta.newest-version")"
tools/osbuilder/scripts/lib.sh:	MUSL_VERSION="$(cat "${kata_versions_file}"  | $yq r -X - "externals.musl.version")"
tools/packaging/scripts/lib.sh:	result=$("${GOPATH}/bin/yq" read -X "$versions_file" "$dependency")
snap/snapcraft.yaml:          branch="$(${yq} r ${versions_file} assets.hypervisor.qemu.architecture.aarch64.branch)"
snap/snapcraft.yaml:          url="$(${yq} r ${versions_file} assets.hypervisor.qemu.url)"
snap/snapcraft.yaml:          commit="$(${yq} r ${versions_file} assets.hypervisor.qemu.architecture.aarch64.commit)"
snap/snapcraft.yaml:          branch="$(${yq} r ${versions_file} assets.hypervisor.qemu.version)"
snap/snapcraft.yaml:          url="$(${yq} r ${versions_file} assets.hypervisor.qemu.url)"
```

(let alone the usage of yq in the tests repository, OTOH, that should use yq 3.4.1)
but at least snap would be good, wouldn't it? since we saw snap failing because of that

**Comment by fgiudici** [comment]
Yep, I think we should take care of yq everywhere... as sooner or later we may want to update to version 4.
And, sure, snap at least should be fixed now.

**Review by Jakob-Naucke — APPROVED** [review]
Okay, I get that. I'm approving since snap is happy. Thanks @fguidici!

**Review by GabyCT — APPROVED** [review]
